src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy

```diff
@@ -0,0 +1,843 @@
+        assert metrics[ACTIVITY_PROCESSED_RULES_FOR_ACCOUNT.formatted(accountId)] == 1
+    }
+
+    def "PBS auction should generate id for bidRequest.(source/imp[0].ext).tid when ext.prebid.createTids=true and transmit activity allowed"() {
+        given: "Bid requests without TID fields and account id"
+        def accountId = PBSUtils.randomNumber as String
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.tap {
+                trace = VERBOSE
+                createTids = true
+            }
+            setAccountId(accountId)
+            imp[0].ext.tid = null
+            source = new Source(tid: null)
+        }
+
+        and: "Activities set with generic bidder allowed"
+        def activities = AllowActivities.getDefaultAllowActivities(TRANSMIT_TID, Activity.defaultActivity)
+
+        and: "Flush metrics"
+        flushMetrics(activityPbsService)
+
+        and: "Save account config with allow activities into DB"
+        def account = getAccountWithAllowActivitiesAndPrivacyModule(accountId, activities)
+        accountDao.save(account)
+
+        when: "PBS processes auction requests"
+        activityPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should generate (source/imp.ext).tid"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+
+        verifyAll {
+            bidderRequest.imp[0].ext.tid
+            bidderRequest.source.tid
+        }
+    }
+
+    def "PBS auction should generate id for bidRequest.(source/imp[0].ext).tid when ext.prebid.createTids=true and transmit activity disallowed"() {
+        given: "Bid requests without TID fields and account id"
+        def accountId = PBSUtils.randomNumber as String
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.tap {
+                trace = VERBOSE
+                createTids = true
+            }
+            setAccountId(accountId)
+            imp[0].ext.tid = null
+            source = new Source(tid: null)
+        }
+
+        and: "Activities set with bidder disallowed"
+        def activity = Activity.getDefaultActivity([ActivityRule.getDefaultActivityRule(Condition.baseCondition, false)])
+        def activities = AllowActivities.getDefaultAllowActivities(TRANSMIT_TID, activity)
+
+        and: "Flush metrics"
+        flushMetrics(activityPbsService)
+
+        and: "Save account config with allow activities into DB"
+        def account = getAccountWithAllowActivitiesAndPrivacyModule(accountId, activities)
+        accountDao.save(account)
+
+        when: "PBS processes auction requests"
+        activityPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should generate (source/imp.ext).tid"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+
+        verifyAll {
+            bidderRequest.imp[0].ext.tid
+            bidderRequest.source.tid
+        }
+    }
```

Comment:
@marki1an I assume that we can simplify code by combining test cases in single one. Same for AMP

```
def "PBS auction should generate id for bidRequest.(source/imp[0].ext).tid when ext.prebid.createTids=#createTidsStatus and transmit activity allow set to #transmitActivityAllowStatus"() {
        given: "Bid requests without TID fields and account id"
        def accountId = PBSUtils.randomNumber as String
        def bidRequest = BidRequest.defaultBidRequest.tap {
            ext.prebid.tap {
                trace = VERBOSE
                createTids = createTidsStatus
            }
            setAccountId(accountId)
            imp[0].ext.tid = null
            source = new Source(tid: null)
        }

        and: "Activities set"
        def activity = Activity.getDefaultActivity([ActivityRule.getDefaultActivityRule(Condition.baseCondition, transmitActivityAllowStatus)])
        def activities = AllowActivities.getDefaultAllowActivities(TRANSMIT_TID, activity)

        and: "Flush metrics"
        flushMetrics(activityPbsService)

        and: "Save account config with allow activities into DB"
        def account = getAccountWithAllowActivitiesAndPrivacyModule(accountId, activities)
        accountDao.save(account)

        when: "PBS processes auction requests"
        activityPbsService.sendAuctionRequest(bidRequest)

        then: "Bidder request should generate (source/imp.ext).tid"
        def bidderRequest = bidder.getBidderRequest(bidRequest.id)

        verifyAll {
            bidderRequest.imp[0].ext.tid
            bidderRequest.source.tid
        }

        and: "Metrics processed across activities should be updated"
        def metrics = activityPbsService.sendCollectedMetricsRequest()
        assert metrics[ACTIVITY_RULES_PROCESSED_COUNT] == 1
        assert metrics[ACTIVITY_PROCESSED_RULES_FOR_ACCOUNT.formatted(accountId)] == 1

        where:
        createTidsStatus | transmitActivityAllowStatus
        null             | true
        true             | true
        true             | false
    }
```

Reply:
We can only merge first two case since in third case won't be populate metrics